orbit/pkg/keystore/keystore_darwin.go

```diff
@@ -0,0 +1,158 @@
+//go:build darwin && cgo
+
+package keystore
+
+/*
+   #cgo LDFLAGS: -framework CoreFoundation -framework Security
+   #include <CoreFoundation/CoreFoundation.h>
+   #include <Security/Security.h>
+*/
+import "C"
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"unsafe"
+)
+
+const service = "com.fleetdm.fleetd.enroll.secret"
+
+var serviceStringRef = stringToCFString(service)
+
+func Exists() bool {
+	return true
+}
+
+func Name() string {
+	return "default keychain"
+}
+
+// AddSecret will add a secret to the keychain. This secret can be retrieved by this application without any user authorization.
+func AddSecret(secret string) error {
+	secret = strings.TrimSpace(secret)
+	if secret == "" {
+		return errors.New("secret cannot be empty")
+	}
+
+	query := C.CFDictionaryCreateMutable(
+		C.kCFAllocatorDefault,
+		0,
+		&C.kCFTypeDictionaryKeyCallBacks,
+		&C.kCFTypeDictionaryValueCallBacks,
+	)
+	defer C.CFRelease(C.CFTypeRef(query))
+
+	data := C.CFDataCreate(C.kCFAllocatorDefault, (*C.UInt8)(unsafe.Pointer(C.CString(secret))), C.CFIndex(len(secret)))
```

Comment:
Something to double check: You may need to `C.free` the string allocated with `C.CString`? (It seems `CFDataCreate` copies the bytes.)

Reply:
Yes, I'll rewrite without the extra memory allocation.